files/views.py

```
@@ -411,39 +411,79 @@ def view_media(request):
+        if not can_see_restricted_media and request.POST.get("password"):
+            if not check_rate_limit(ip, media.friendly_token):
+                rate_limited = True
+            else:
+                submitted_password = request.POST.get("password")
+                if check_password(submitted_password, media.password):
+                    can_see_restricted_media = True
+                    try:
+                        token = generate_token(media_uid)
+                        request.session[f"media_token_{media.friendly_token}"] = token
+                        media_access_token = token
+                    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)